File: LeetCode_Solutions/4-median-of-two-sorted-arrays/median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        merge=[]
        i,j=0,0
        while i<len(nums1) and j<len(nums2):
            if nums1[i]<nums2[j]:
                merge.append(nums1[i])
                i+=1
            else:
                merge.append(nums2[j])
                j+=1
        while i<len(nums1):
            merge.append(nums1[i])
            i+=1
        while j<len(nums2):
            merge.append(nums2[j])
            j+=1
        #print(merge)
        n=len(merge)
        if n%2!=0:
            return merge[n//2]
        else:
            mid=n//2
            ans=(merge[mid-1]+merge[mid])/2
            return ans

File: LeetCode_Solutions/4-median-of-two-sorted-arrays/median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
         int[] merge = new int[nums1.length + nums2.length];
        int i = 0, j = 0, k = 0;
        
        while (i < nums1.length && j < nums2.length) {
            if (nums1[i] < nums2[j]) {
                merge[k++] = nums1[i++];
            } else {
                merge[k++] = nums2[j++];
            }
        }
        
        while (i < nums1.length) {
            merge[k++] = nums1[i++];
        }
        
        while (j < nums2.length) {
            merge[k++] = nums2[j++];
        }
        
        int n = merge.length;
        if (n % 2 != 0) {
            return merge[n / 2];
        } else {
            int mid = n / 2;
            return (merge[mid - 1] + merge[mid]) / 2.0;
    }
}
}